web/src/main/java/com/epam/esm/config/DispatcherServletInitializer.java

```diff
@@ -0,0 +1,32 @@
+package com.epam.esm.config;
+
+import org.springframework.web.context.WebApplicationContext;
+import org.springframework.web.servlet.DispatcherServlet;
+import org.springframework.web.servlet.FrameworkServlet;
+import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer;
+
+public class DispatcherServletInitializer extends AbstractAnnotationConfigDispatcherServletInitializer {
+    private static final String ROOT_MAPPING = "/";
+
+    @Override
+    protected Class<?>[] getRootConfigClasses() {
+        return new Class[] {};
+    }
+
+    @Override
+    protected Class<?>[] getServletConfigClasses() {
+        return new Class[] { ApplicationConfig.class };
+    }
+
+    @Override
+    protected String[] getServletMappings() {
+        return new String[] { ROOT_MAPPING };
+    }
+
+    @Override
+    protected FrameworkServlet createDispatcherServlet(WebApplicationContext servletAppContext) {
+        DispatcherServlet dispatcher = (DispatcherServlet) super.createDispatcherServlet(servletAppContext);
+        dispatcher.setThrowExceptionIfNoHandlerFound(true);
```

Comment:
We need this flag being `true` to make servlet container generate exception in case when no URL mapping found. This exception is caught in `ApplicationExceptionHandler`.

Reply:
Application falls back to Tomcat error page (attached screenshot). I wanted to produce meaningful JSON response for the case.
![image](https://user-images.githubusercontent.com/42818718/135139057-39c5b19f-8794-442e-afd0-53f4e44e0553.png)